File: 1786-count-the-number-of-consistent-strings/1786-count-the-number-of-consistent-strings.cpp
#pragma GCC optimize(O3, "unroll-loops")
#pragma GCC target ("avx")
#pragma GCC target ("-fsplit-loops")

class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        int ans = 0;
        bool stat, c_stat;
        for (string wrd: words) {
            stat = true;
            for (char c: wrd) {
                c_stat = false;
                for (char ref: allowed) {
                    c_stat = ref == c;
                    if (c_stat) break;
                }
                if (!c_stat) {
                    stat = false;
                    break;
                }
            }

            if (stat) ans++;
        }

        return ans;
    }
};

auto init = []() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    return 'c';
}();

File: 1786-count-the-number-of-consistent-strings/1786-count-the-number-of-consistent-strings.kt
class Solution {
    fun countConsistentStrings(allowed: String, words: Array<String>): Int {
        val mp: BooleanArray = BooleanArray(26) { false }.apply {
            for (c in allowed) set(c - 'a', true);
        }
        var ans: Int = 0; var stat: Boolean;
        for (str in words) {
            stat = true;
            for (c in str) 
                if (!mp[c - 'a']) {
                    stat = false;
                    break;
                }

                if (stat) ans++;
        }
        return ans; 
    }
}